MirahezeBots/plugins/wikimgnt.py

```diff
@@ -0,0 +1,288 @@
+"""wikimgnt.py - interact with MediaWiki"""
+
+import requests
+import re
+import time
+import random
+import json
+
+from sopel.module import rule, commands, example
+from sopel.config.types import StaticSection, ValidatedAttribute, ListAttribute
+
+
+class WikimgntSection(StaticSection):
+    log_wiki_url = ValidatedAttribute('log_wiki_url', str)
+    log_page = ValidatedAttribute('log_page', str)
+    wiki_acl = ListAttribute('wiki_acl')
+    wiki_farm = ValidatedAttribute('wiki_farm', bool)
+    wiki_domain = ValidatedAttribute('wiki_domain', str)
+    wiki_username = ValidatedAttribute('wiki_username', str)
+    wiki_password = ValidatedAttribute('wiki_password', str)
+
+
+def setup(bot):
+    bot.config.define_section('wikimgnt', WikimgntSection)
+
+
+def configure(config):
+    config.define_section('wikimgnt', WikimgntSection, validate=False)
+    config.wikimgnt.configure_setting('log_wiki_url', 'What is the URL of the wiki that you would like .log messages to go to? Please specify the URL to that wikis api.php.')
+    config.wikimgnt.configure_setting('log_page', 'What page on that wiki would like .log messages to go to? Instead of a space, type a _ please.')
+    config.wikimgnt.configure_setting('wiki_acl', 'Please enter NickServ accounts that are allowed to use the commands in this module. (No spaces)')
+    config.wikimgnt.configure_setting('wiki_farm', 'Are you using this for a wiki farm? (true/false)')
```

Comment:
If not false: link to a central wiki discover api and then use that to generate a sites list?

Reply:
I'm debating how to go about multi wiki support. I'll work on this at some point.

Reply:
Multi wiki support works for everything but .log, if you set wiki_farm to true

Reply:
I'll get my head round something later in the week, still debating some stuff